engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/NkGlobalParameters.scala

```diff
@@ -10,25 +9,32 @@ import org.apache.flink.api.common.ExecutionConfig.GlobalJobParameters
 import pl.touk.nussknacker.engine.api.ProcessVersion
 import net.ceedubs.ficus.readers.ArbitraryTypeReader._
 import net.ceedubs.ficus.Ficus._
+import pl.touk.nussknacker.engine.api.deployment.DeploymentData
 
 //we can use this class to pass config through RuntimeContext to places where it would be difficult to use otherwise
 //Also, those configuration properties will be exposed via Flink REST API/webconsole
 case class NkGlobalParameters(buildInfo: String,
                               processVersion: ProcessVersion,
+                              deploymentData: DeploymentData,
                               configParameters: Option[ConfigGlobalParameters],
                               namingParameters: Option[NamingParameters]) extends GlobalJobParameters {
 
   //here we decide which configuration properties should be shown in REST API etc.
   //For now it will be only deployment information
-  //NOTE: this information is used in FlinkRestManager - any changes here should be reflected there
+  //NOTE: some of the information is used in FlinkRestManager - any changes here should be reflected there
   override def toMap: util.Map[String, String] = {
-    //we wrap in HashMap because .asJava creates not-serializable map in 2.11
-    new util.HashMap(Map[String, String](
+    val deploymentInfo = deploymentData.additionalDeploymentData.map {
+      case (k, v) => s"deployment.properties.$k" -> v
```

Comment:
I wonder how to handle situation when it will came something big in those properties? WDYT? I know that we have the same problem in `configMap` but there are rather static values and here someone can put e.g. something generated on manager site. Maybe we should extract protected method like `extractDeploymentDataToMap`? But still we have case class that shouldn't be extendend. Maybe we should have `AbstractNkGlobalParameters` and `NkGlobalParameters` as a standard implementation?

Reply:
I extracted part of serialization to strategy.